### PR TITLE
Sql fixes

### LINKED ISSUE #689: HSQL database is wide open
In WebGoat v8.0.0.M25, the HSQL database listens on TCP port 9001, not bound to any address.  Any HSQL client on the Internet can connect to it with full read-write privileges without any authentication.  While WebGoat is a web application with intentional security holes, surely this security hole is a bit over the top?

See [`HSQLDBDatabaseConfig.hsqlStandalone()`](https://github.com/WebGoat/WebGoat/blob/v8.0.0.M25/webgoat-server/src/main/java/org/owasp/webgoat/HSQLDBDatabaseConfig.java#L28-L41).  I suggest calling [`.setAddress()`](http://hsqldb.org/doc/2.0/apidocs/org/hsqldb/server/Server.html#setAddress-java.lang.String-) so that the socket can be configured to bind to localhost.  Alternatively, set a password for connecting to the database.

### PR DESCRIPTION
**Pull request by nbaars**
- Moved database scripts to lessons 
- Split core WebGoat and lesson data into 2 different DB schemas
- Updated lessons according to bug 
- Bind HSQLDB to same address as WebGoat, fixes #689
- Fix for #666
- Fix for #665
- Fix for #664
- Fix for #663
- Partial fix for #662 (we are not supporting ")

### CODE REVIEW

**Comment by zubcevic** [comment]
By the way, a lot of great work which will definitely make it more robust. Especially in the way it can clean the old challenges related stuff. Just consider to add the migration baseline stuff for existing databases.

File: webgoat-container/src/main/java/org/owasp/webgoat/DatabaseInitialization.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Bean(initMethod = "migrate")
+    @DependsOn("flyWayContainer")
+    public Flyway flywayLessons() {
+        return Flyway
+                .configure()
+                .configuration(Map.of("driver", driverClassName))
+                .dataSource(dataSource)
+                .load();
```
[the remark this answers is not in the document]
Comment (nbaars) [reply webgoat-container/src/main/java/org/owasp/webgoat/DatabaseInitialization.java]: yes but that will not work, the old tables from WebGoat container will remain in the PUBLIC schema. This is a breaking change which does not matter as each released version has it's own .webgoat directory so empty by default. In the next release we can then apply Flyway with new scripts to upgrade an old version

[the remark this answers is not in the document]
Comment (nbaars) [reply webgoat-container/src/main/java/org/owasp/webgoat/DatabaseInitialization.java]: so support users moving from one version to the next and leaving the data inside the database, something we now do not support

[the remark this answers is not in the document]
Comment (zubcevic) [reply webgoat-container/src/main/java/org/owasp/webgoat/DatabaseInitialization.java]: ok

File: webgoat-integration-tests/src/test/java/org/owasp/webgoat/SqlInjectionLessonTest.java [hunk 2]
```diff
@@ -39,14 +39,14 @@ public void runTests() {
...
 
-        params.clear();
-        params.put("query", sql_4_drop);
-        checkAssignment(url("/WebGoat/SqlInjection/attack4"), params, false);
-
         params.clear();
         params.put("query", sql_4_add);
         checkAssignment(url("/WebGoat/SqlInjection/attack4"), params, true);
 
+        params.clear();
```
[the remark this answers is not in the document]
Comment (nbaars) [reply webgoat-integration-tests/src/test/java/org/owasp/webgoat/SqlInjectionLessonTest.java]: I will have to look into this, I cannot remember I put it here and for what reason :-)

**Pull request merged**